### ISSUE TITLE
[task plugin] multiple line fields are appended with \r

### ISSUE THREAD

**Issue by srinivasupadhya** [gocd/gocd#815]
```
<task>
                <pluginConfiguration id="script-executor" version="1" />
                <configuration>
                  <property>
                    <key>script</key>
                    <value>ls -lah&#xD;
echo "Hey! $GO_TRIGGER_USER"&#xD;
echo "version - GO_VERSION"</value>
                  </property>
                </configuration>
                <runif status="passed" />
              </task>
```

I did some analysis on this. `&#xD;` is carriage return character. I do not know how thats getting inserted in task plugin field.
When &#xD; is read & sent across to task plugin on agent side the file contents are:
ls -lah^M$
echo "Hello $GO_TRIGGER_USER"^M$
cat -A *.sh

And when i manually remove &#xD; from config xml the file contents are:
ls -lah$
echo "Hello $GO_TRIGGER_USER"$
cat -A *.sh

**Comment by arvindsv** [comment]
Closing as stale.

**Comment by nush-1qb** [comment]
Was able to reproduce this on macOS Mojave 10.14.3

Pretty much any task that is defined through the UI will have a `\r` character anywhere I have a new line.

Simply using the built-in command for docker build and then saving the task, and then downloading the yaml or json shows the characters added at the end.

For instance, if I want to define a task to build a docker image, this is what I will put in the text field:
```
-c
docker build -t #{CONTAINER_REGISTRY}/#{IMAGE_NAME}:#{IMAGE_TAG} #{BUILD_DIR} -f #{DOCKERFILE_NAME}
```

and this is what the .gocd.yaml file shows:
```
- exec:
    arguments:
    - "-c\r"
    - 'docker build -t #{CONTAINER_REGISTRY}/#{IMAGE_NAME}:#{IMAGE_TAG}
                  #{BUILD_DIR} -f #{DOCKERFILE_NAME}'
    command: /bin/bash
    run_if: passed
```

This is extremely annoying, as I can't successfully create a task through the UI. Is there a workaround?

**Comment by arvindsv** [comment]
Assuming you're on a newer version of GoCD, https://github.com/gocd/gocd/issues/7557 is the more relevant issue and it was fixed in GoCD 20.1.0, released last week.
